core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/naming/OicrSampleAliasGenerator.java

```diff
@@ -33,10 +33,9 @@ public String generateName(Sample t) {
       throw new IllegalArgumentException("Can only generate an alias for detailed samples");
     }
     SampleAdditionalInfo detailed = (SampleAdditionalInfo) t;
-    
-    for (SampleAdditionalInfo parent = (SampleAdditionalInfo) detailed.getParent(); parent != null;
-        parent = (SampleAdditionalInfo) parent.getParent()) {
-      if (isAnalyteAliquot(parent)) {
+
+    for (SampleAdditionalInfo parent = detailed.getParent(); parent != null; parent = parent.getParent()) {
+      if (isAliquotSample(parent) || isStockSample(parent)) {
```

Comment:
Your earlier change here was correct. The condition should be `if (isAliquotSample(parent))`. In the case that the direct parent is a stock, it should be ignored for alias derivation purposes - the aliquot part is appended to the tissue's alias